### ISSUE TITLE
Failing to build {chem}[foss/2021a] RDKit v2023.03.3 w/ Python 3.9.5

### ISSUE THREAD

**Issue by grvlbit** [easybuilders/easybuild-easyconfigs#19413]
I am facing a strange architecture dependent issue trying to build [RDKit-2023.03.3-foss-2021a.eb](https://github.com/easybuilders/easybuild-easyconfigs/blob/ed6b43bae472a571b7c0baf0276a015273fddc15/easybuild/easyconfigs/r/RDKit/RDKit-2023.03.3-foss-2021a.eb) on AMD EPYC Zen2. During the `ctest` part of the build step the compilation is stopped due to failing tests:

```
99% tests passed, 2 tests failed out of 214

Total Test time (real) = 281.48 sec

The following tests FAILED:
        124 - testMMFFForceFieldHelpers (Subprocess aborted)
        126 - testUFFForceFieldHelpers (Subprocess aborted)
Errors while running CTest
 (at easybuild/tools/run.py:681 in parse_cmd_output)
== 2023-12-13 13:19:00,116 build_log.py:267 INFO ... (took 7 mins 28 secs)
== 2023-12-13 13:19:00,116 filetools.py:2012 INFO Removing lock /storage/homefs/l_easybuild/easybuild/software/.locks/_storage_homefs_l_easybuild_easybuild_software_RDKit_2023.03.3-foss-2021a.lock...
== 2023-12-13 13:19:00,117 filetools.py:383 INFO Path /storage/homefs/l_easybuild/easybuild/software/.locks/_storage_homefs_l_easybuild_easybuild_software_RDKit_2023.03.3-foss-2021a.lock successfully removed.
== 2023-12-13 13:19:00,117 filetools.py:2016 INFO Lock removed: /storage/homefs/l_easybuild/easybuild/software/.locks/_storage_homefs_l_easybuild_easybuild_software_RDKit_2023.03.3-foss-2021a.lock
== 2023-12-13 13:19:00,117 easyblock.py:4277 WARNING build failed (first 300 chars): cmd " make  -j 128 && cp -RT /dev/shm/RDKit/2023.03.3/foss-2021a/rdkit-*/ ./ && export RDBASE=$PWD && export PYTHONPATH=$PWD:$PYTHONPATH && export LD_LIBRARY_PATH=/dev/shm/RDKit/2023.03.3/foss-2021a/easybuild_obj/lib:${LD_LIBRARY_PATH} && ctest --output-on-failure" exited with exit code 8 and output
== 2023-12-13 13:19:00,117 easyblock.py:328 INFO Closing log for application name RDKit version 2023.03.3
```

I've tried limiting the number of threads during the compilation using `--parallel` and interestingly this sometimes leads to more failing tests. The same build succeeds without issues on an older Intel broadwell node! 
Any ideas what could cause the tests to fail specifically on zen2 nodes?

### eb info
```
eb --version
This is EasyBuild 4.8.2 (framework: 4.8.2, easyblocks: 4.8.2) on host bnode001.
``` 

### system info

```

             //////////               l_easybuild@bnode001
          (/(/(/(/(/(/(/(/            OS: Rocky Linux 9.3
      ,//////////////////////         Kernel: x86_64 Linux 5.14.0-362.8.1.el9_3.x86_64
    (/(/(/(/(/(/(/(/(/(/(/(/(/(*      
   //////////////////////////////     
  (/(/(/(/(/(/(/(/(/(///(/(/(/(/(/    Shell: bash 5.1.8
 ///////////////////      /////////   
 /(/(/(/(/(/(/(/(/          //(/(/(   CPU: AMD EPYC 7742 64-Core @ 128x 2.25GHz
 //////////////                ////   GPU: ASPEED Technology, Inc. ASPEED Graphics Family (rev 41)
  (/(/(/(/(//        /(/(        /    RAM: 32035MiB / 1031386MiB
   ///////        .////////.
    //(/        (/(/(/(/(/(/(/
             .///////////////
           /(/(/(/(/(/(/(,
             //////////
```

**Comment by boegel** [comment]
@grvlbit Is there any additional information in the ctest output file that's a bit more specific than "`Subprocess aborted`"?

My first guess would be that you're hitting some kind of memory limit.

Specifying `--parallel` will help for the build, but may not affect the number of cores that `ctest` thinks it can use...

**Comment by grvlbit** [comment]
Sorry for the late reply. I took some time away from the computer.

I was unable to get more details on the ctest output. I agree that "`Subprocess aborted`" sounds like some memory limit. Funny enough, I was able to build the module on the same architecture once I restricted the available memory during the build within a SLURM job.

For us this issue is resolved (since we were able to build within SLURM), but I don't know if you want to keep this open?
